e2etest/bgptests/infra_setup.go

```diff
@@ -66,79 +71,106 @@ func InfraSetup(ipv4Addresses, ipv6Addresses []string, cs *clientset.Clientset)
+			HostIPv4: hostIPv4,
+			HostIPv6: hostIPv6,
 		},
-		Router: frrconfig.RouterConfig{
-			ASN:      metalLBASN,
-			BGPPort:  179,
-			Password: "ibgp-test",
+		{
+			Name: "ibgp-multi-hop",
+			Neighbor: frrconfig.NeighborConfig{
+				ASN:      metalLBASN,
+				Password: "ibgp-test",
+				MultiHop: true,
+			},
+			Router: frrconfig.RouterConfig{
+				ASN:      metalLBASN,
+				BGPPort:  180,
+				Password: "ibgp-test",
+			},
+			Network:  multiHopNetwork,
+			HostIPv4: hostIPv4,
+			HostIPv6: hostIPv6,
 		},
-		Network:  containersNetwork,
-		HostIPv4: hostIPv4,
-		HostIPv6: hostIPv6,
-	}
-	ibgpMultiHopContainerConfig := frrcontainer.Config{
-		Name: "ibgp-multi-hop",
-		Neighbor: frrconfig.NeighborConfig{
-			ASN:      metalLBASN,
-			Password: "ibgp-test",
-			MultiHop: true,
+		{
+			Name: "ebgp-multi-hop",
+			Neighbor: frrconfig.NeighborConfig{
+				ASN:      metalLBASN,
+				Password: "ebgp-test",
+				MultiHop: true,
+			},
+			Router: frrconfig.RouterConfig{
+				ASN:      externalASN,
+				BGPPort:  180,
+				Password: "ebgp-test",
+			},
+			Network:  multiHopNetwork,
+			HostIPv4: hostIPv4,
+			HostIPv6: hostIPv6,
 		},
-		Router: frrconfig.RouterConfig{
-			ASN:      metalLBASN,
-			BGPPort:  180,
-			Password: "ibgp-test",
-		},
-		Network:  multiHopNetwork,
-		HostIPv4: hostIPv4,
-		HostIPv6: hostIPv6,
-	}
-	ebgpMultiHopContainerConfig := frrcontainer.Config{
-		Name: "ebgp-multi-hop",
-		Neighbor: frrconfig.NeighborConfig{
-			ASN:      metalLBASN,
-			Password: "ebgp-test",
-			MultiHop: true,
-		},
-		Router: frrconfig.RouterConfig{
-			ASN:      externalASN,
-			BGPPort:  180,
-			Password: "ebgp-test",
-		},
-		Network:  multiHopNetwork,
-		HostIPv4: hostIPv4,
-		HostIPv6: hostIPv6,
-	}
-	ebgpSingleHopContainerConfig := frrcontainer.Config{
-		Name:    "ebgp-single-hop",
-		Network: containersNetwork,
-		Neighbor: frrconfig.NeighborConfig{
-			ASN:      metalLBASN,
-			MultiHop: false,
-		},
-		Router: frrconfig.RouterConfig{
-			ASN:     externalASN,
-			BGPPort: 179,
+		{
+			Name:    "ebgp-single-hop",
+			Network: containersNetwork,
+			Neighbor: frrconfig.NeighborConfig{
+				ASN:      metalLBASN,
+				MultiHop: false,
+			},
+			Router: frrconfig.RouterConfig{
+				ASN:     externalASN,
+				BGPPort: 179,
+			},
 		},
 	}
 
 	var res []*frrcontainer.FRR
 	var err error
-	if containersNetwork == "host" {
-		res, err = frrcontainer.Create(ibgpSingleHopContainerConfig)
+	if containers != "" {
+		err := validateContainersNames(containers)
+		if err != nil {
+			return nil, err
+		}
+
+		configs, err := configsFor(containers, frrContainers)
+		if err != nil {
+			return nil, err
+		}
+
+		res, err = frrcontainer.ConfigureExisting(configs...)
+		if err != nil {
+			return nil, err
+		}
+
+		if containsMultiHop(res) {
+			err = multiHopSetUp(res, cs)
+			if err != nil {
+				return nil, err
+			}
+		}
+	} else if containersNetwork == "host" {
+		res, err = frrcontainer.Create(frrContainers[ibgpSingleHopContainerConfig])
```

Comment:
see the comment above about the slice. It's clearer if you refer the container by name (we could rename the index as something like ...Index, but I still think there's no value in converting the single instances to a slice)

Reply:
done.